### ISSUE TITLE
AssertionError: unknown FUN_DECL

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#19]
The function
```
fun wordcount lines =
  let
    fun split0 [] word words = word :: words
      | split0 (#" " :: s) word words = split0 s "" (word :: words)
      | split0 (c :: s) word words = split0 s (word ^ (String_str c)) words
    fun split s = List_rev (split0 (String_explode s) "" [])
  in
    from line in lines,
        word in split(line)
    group word compute count of 1 as count
  end;
```
gives error
```
Exception in thread "main" java.lang.AssertionError: unknown FUN_DECL; fun split0 ([]) word words = word :: words | split0 (  :: s) word words = split0 s "" (word :: words) | split0 (c :: s) word words = split0 s (word ^ String_str c) words
	at net.hydromatic.morel.compile.Compiler.compileDecl(Compiler.java:327)
	at net.hydromatic.morel.compile.Compiler.compileLet(Compiler.java:301)
```

Also note that in the error message, `split0 (  :: s) word words` should be `split0 (#" "  :: s) word words`.

**Comment by julianhyde** [comment]
Simpler case:
```
fun foo arg =
  let
    fun baz x = x
  in
    baz arg
  end
```

**Comment by julianhyde** [comment]
Fixed in b988f3c62fe20efe0ba1b20661fbebd4c829b36e.
